backend-test/TestHelpers.hs

```diff
@@ -34,125 +36,56 @@ import System.Directory (removeFile)
+        , rqPaths           = splitPath reUri
+        , rqUri             = reUri
         , rqQuery           = ""
-        , rqInputsQuery     = [("name", Input {
-            inputValue = Right (BSL8.pack courseName),
-            inputFilename = Nothing,
-            inputContentType = defaultContentType
-          })]
-        , rqInputsBody      = inputsBody
+        , rqInputsQuery     = map (fmap convertInput) queryInputs
+        , rqInputsBody      = reInputsBody
         , rqCookies         = []
         , rqVersion         = HttpVersion 1 1
         , rqHeaders         = Map.empty
-        , rqBody            = requestBody
+        , rqBody            = reBody
         , rqPeer            = ("127.0.0.1", 0)
         }
-
--- | A mock request for running ServerPartWithCourseInfoQuery, specifically for courseInfo
-mockRequestWithCourseInfoQuery :: String -> IO Request
-mockRequestWithCourseInfoQuery dept = do
-    inputsBody <- newMVar []
-    requestBody <- newEmptyMVar
-    return Request
-        { rqSecure          = False
-        , rqMethod          = GET
-        , rqPaths           = ["course-info"]
-        , rqUri             = "/course-info"
-        , rqQuery           = ""
-        , rqInputsQuery     = [("dept", Input {
-            inputValue = Right (BSL8.pack dept),
-            inputFilename = Nothing,
-            inputContentType = defaultContentType
-          })]
-        , rqInputsBody      = inputsBody
-        , rqCookies         = []
-        , rqVersion         = HttpVersion 1 1
-        , rqHeaders         = Map.empty
-        , rqBody            = requestBody
-        , rqPeer            = ("127.0.0.1", 0)
+  where
+    -- | Helper to convert a String to a query input parameter
+    convertInput :: String -> Input
+    convertInput paramValue = Input
+        { inputValue = Right (BSL8.pack paramValue)
+        , inputFilename = Nothing
+        , inputContentType = defaultContentType
         }
 
--- | A mock request for running ServerPartWithProgramQuery, specifically for retrieveProgram
-mockRequestWithProgramQuery :: String -> IO Request
-mockRequestWithProgramQuery programCode = do
-    inputsBody <- newMVar []
-    requestBody <- newEmptyMVar
-    return Request
-        { rqSecure          = False
-        , rqMethod          = GET
-        , rqPaths           = ["program"]
-        , rqUri             = "/program"
-        , rqQuery           = ""
-        , rqInputsQuery     = [("code", Input {
-            inputValue = Right (BSL8.pack programCode),
-            inputFilename = Nothing,
-            inputContentType = defaultContentType
-          })]
-        , rqInputsBody      = inputsBody
-        , rqCookies         = []
-        , rqVersion         = HttpVersion 1 1
-        , rqHeaders         = Map.empty
-        , rqBody            = requestBody
-        , rqPeer            = ("127.0.0.1", 0)
-        }
+    -- | Split a string when a predicate is true
```

Comment:
It's great you implemented this as an exercise in Haskell, but you can use a function from one of our dependencies instead: https://hackage-content.haskell.org/package/MissingH-1.6.0.2/docs/Data-List-Utils.html#v:split